src/main/java/crazypants/enderio/conduit/AbstractConduit.java

```diff
@@ -124,25 +126,29 @@ protected NBTTagCompound getNbtRootForType(NBTTagCompound nbt, boolean createIfN
   }
 
   @Override
-  public ConnectionMode getConnectionMode(EnumFacing dir) {
+  @Nonnull
+  public ConnectionMode getConnectionMode(@Nonnull EnumFacing dir) {
     ConnectionMode res = conectionModes.get(dir);
     if (res == null) {
       return getDefaultConnectionMode();
     }
     return res;
   }
 
+  @Nonnull
   protected ConnectionMode getDefaultConnectionMode() {
     return ConnectionMode.IN_OUT;
   }
 
   @Override
-  public List<ItemStack> getDrops() {
-    return Collections.singletonList(createItem());
+  @Nonnull
+  public NNList<ItemStack> getDrops() {
+    // return Collections.singletonList(createItem());
+    return new NNList<ItemStack>(1, ItemStack.EMPTY);
```

Comment:
Sure you want to return a list of one empty itemstack? If not: NNList.emptyList()